cloudinit/ssh_util.py

```diff
@@ -544,11 +544,27 @@ def parse_ssh_config_map(fname):
     return ret
 
 
+def _includes_dconf(fname: str) -> bool:
+    if not os.path.isfile(fname):
+        return False
+    with open(fname, "r") as f:
+        for line in f:
+            if line.startswith(f"Include {fname}.d/*.conf"):
+                return True
+    return False
+
+
 def update_ssh_config(updates, fname=DEF_SSHD_CFG):
     """Read fname, and update if changes are necessary.
 
     @param updates: dictionary of desired values {Option: value}
     @return: boolean indicating if an update was done."""
+    if _includes_dconf(fname):
+        if not os.path.isdir(f"{fname}.d"):
+            util.ensure_dir(f"{fname}.d", mode=0o755)
+        fname = os.path.join(f"{fname}.d", "99-cloud-init.conf")
```

Comment:
reading now again [man sshd_config](https://manpages.ubuntu.com/manpages/jammy/man5/sshd_config.5.html):

> /etc/ssh/sshd_config.d/*.conf files are included at the start of the configuration file, so options set there will override those in /etc/ssh/sshd_config

and

> For each keyword, the first obtained value will be used.

Doesn't that mean that the filename should be `00-cloud-init.conf` ? Because the values set by cloud-init need to be the **first** values read to make sure those values are used.

Reply:
Correct. Thanks for pointing this out. I updated it to give more preference over our conf.